api-service/crowdsource-webapp/src/app/relative-toxicity-job/relative-toxicity-job.component.html

```diff
@@ -0,0 +1,88 @@
+<div class="container">
+  <h1>Comparative Toxicity Job</h1>
+
+  <div class="counts" *ngIf="notEmbedded">
+    <div>my worker id: {{userNonce}}</div>
+    <div>local_sent_count: {{local_sent_count}}</div>
+    <div>training_answer_count: {{training_answer_count}}</div>
+    <div>user_mean_score: {{user_mean_score}}</div>
+    <div>overall_job_answer_count: {{overall_job_answer_count}}</div>
+    <div>overall_job_mean_score: {{overall_job_mean_score}}</div>
+  </div>
+
+  <div *ngIf="errorMessages.length > 0" class="errorMessage">
+    <div>
+      <div *ngFor="let errorMessage of errorMessages">{{errorMessage}}</div>
+    </div>
+    <button mat-icon-button color="warn" (click)="clearError()"><mat-icon>close</mat-icon></button>
+  </div>
+
+  <div *ngIf="!question">
+    Loading...
+  </div>
+
+  <div *ngIf="question">
+    <div class="id" *ngIf="notEmbedded">Question Id: <span>{{question.id}}</span></div>
+
+    <div class="experimentToggle">
+      <mat-slide-toggle labelPosition="before "[(ngModel)]="useRadioButtons">Use radio buttons</mat-slide-toggle>
+    </div>
+
+    <div *ngIf="useRadioButtons; else dragAndDrop">
+      <div class="text-header">Comment A</div>
+      <div class="text">{{question.comment_a}}</div>
+      <div class="text-header">Comment B</div>
+      <div class="text">{{question.comment_b}}</div>
+      <div class="question-group">
+        <div class="toxicity-selected-value">
+          <span class="label-heading">How does comment A compare to comment B?</span>
+        </div>
+        <mat-radio-group class="radioGroup" [(ngModel)]="toxicityOption">
+          <mat-radio-button *ngFor="let option of toxicityOptions" class="toxicity-radio-button" [value]="option">{{option}}</mat-radio-button>
+        </mat-radio-group>
+      </div>
+    </div>
+
+    <ng-template #dragAndDrop>
```

Comment:
The functionality for the drag/drop is very orthogonal to the functionality for the radio-buttons; I think this would probably be much better separated into two separate components that each handle their functionality rather than have this merged thing in one component. It'll simplify understanding of each component also. e.g. drag function is not relevant to radio-buttons.

Reply:
Hmm...I agree in theory but I do think that it comes down to whether we want the component level distinctions to be on a per-job basis or per-UI basis (I'm thinking of what makes sense routing/URL wise). We've been naming them to be on a per-job basis so far. The component isn't very complex, so I'm not sure the complexity of having to create two components for the same job is worth exchanging for the complexity of having two UI options in a given component. I'm going to think about this some more.

Reply:
I wasn't thinking to change the URL routes, but just put the encapsulated functionality into it's own component as the overlap of messier and more complex. i.e. creating two sub-components one for each kind of view, and the top component is then the switch + binding to top level values to inputs, or something like that. But fine to add a comment and leave as is for now.

Reply:
Yes, that's a good point. I'll add a comment and make this change next week.